app/controllers/api/base_controller.rb

```diff
@@ -264,6 +244,68 @@ def action_permission
       end
     end
 
+    def parent_permission(child_permission)
+      case child_permission.to_s
+      when 'create', 'destroy'
+        'edit'
+      when 'edit', 'view'
+        'view'
+      else
+        raise ::Foreman::Exception.new(N_("unknown parent permission for %s"), "#{params[:controller]}##{child_permission}")
+      end
+    end
+
+    def parent_resource_details
+      classes_arr = params.map { |key, value| extract_resource_from_param(key, value) }
```

Comment:
You pass value here only to get it back in the array, but it's not needed inside `extract_resource_from_param` method. I think that `extract_resource_from_param` should only return the resource class for the key and the construction should be out of it. Later you take just the first record anyway. So something like

```ruby
def parent_resource_details
  params.find do |param, value| 
    parameter_id = value
    parent_name, parent_class = extract_resource_from_param(param)
  end

  return nil if parent_name.nil?
  parent_scope = scope_for(parent_class, :permission => "#{parent_permission(action_permission)}_#{parent_name.pluralize}")
  parent_scope = select_by_resource_id_scope(parent_scope, parent_class, parent_id)
  [parent_name, parent_scope]
end

def extract_resource_from_param(param)
  md = param.match(/(\w+)_id$/)
  md ? [md[0], resource_class_for(resource_name(md[1])] : nil
end
```